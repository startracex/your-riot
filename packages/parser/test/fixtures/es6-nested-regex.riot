<es6-nested-regex>
  <h1>{ message }</h1>

  <script>
    const x = 1

    export default {
      message: `
        regex: ${
          /x}`/.source
        }`,

      foobar() { `
        foobar() {
          return ${
            foo( 5 /x/.1 )  // not regex
          }
        }`
      }
    }





  function foo(bar) {

  }

  function foo(bar) {

  }

  </script>

  <style>
    :root {
      color: blue;
    }
  </style>
</es6-nested-regex>
