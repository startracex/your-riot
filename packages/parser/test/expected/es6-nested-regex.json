{
  "template": {
    "type": 1,
    "name": "es6-nested-regex",
    "start": 0,
    "end": 450,
    "isCustom": true,
    "nodes": [
      {
        "type": 1,
        "name": "h1",
        "start": 21,
        "end": 41,
        "nodes": [
          {
            "type": 3,
            "text": "{ message }",
            "start": 25,
            "end": 36,
            "expressions": [
              {
                "text": " message ",
                "start": 25,
                "end": 36
              }
            ],
            "parts": [
              "message"
            ]
          }
        ]
      }
    ]
  },
  "css": {
    "type": 1,
    "name": "style",
    "start": 375,
    "end": 430,
    "text": {
      "type": 3,
      "text": "\n    :root {\n      color: blue;\n    }\n  ",
      "start": 382,
      "end": 422
    }
  },
  "javascript": {
    "type": 1,
    "name": "script",
    "start": 45,
    "end": 371,
    "text": {
      "type": 3,
      "text": "\n    const x = 1\n\n    export default {\n      message: `\n        regex: ${\n          /x}`/.source\n        }`,\n\n      foobar() { `\n        foobar() {\n          return ${\n            foo( 5 /x/.1 )  // not regex\n          }\n        }`\n      }\n    }\n\n\n\n\n\n  function foo(bar) {\n\n  }\n\n  function foo(bar) {\n\n  }\n\n  ",
      "start": 53,
      "end": 362
    }
  }
}